packages/jaeger-ui/src/model/trace-viewer.tsx

```diff
@@ -44,10 +49,59 @@ export function _getTraceNameImpl(spans: Span[]) {
       candidateSpan = spans[i];
     }
   }
-  return candidateSpan ? `${candidateSpan.process.serviceName}: ${candidateSpan.operationName}` : '';
+
+  if (!candidateSpan) {
+    return null;
+  }
+
+  return {
+    serviceName: candidateSpan.process.serviceName,
+    operationName: candidateSpan.operationName,
+  };
 }
 
-export const getTraceName = _memoize(_getTraceNameImpl, (spans: Span[]) => {
+export const getTracePageHeaderParts = _memoize(_getTracePageHeaderPartsImpl, (spans: Span[]) => {
   if (!spans.length) return 0;
   return spans[0].traceID;
 });
+
+export function getTraceName(spans: Span[]): string {
+  const tracePageHeaderParts = getTracePageHeaderParts(spans);
+
+  return tracePageHeaderParts
+    ? `${tracePageHeaderParts.serviceName}: ${tracePageHeaderParts.operationName}`
+    : '';
+}
+
+export function getTracePageTitle(spans: Span[]): string {
+  const tracePageHeaderParts = getTracePageHeaderParts(spans);
+
+  return tracePageHeaderParts
+    ? `${tracePageHeaderParts.operationName} (${tracePageHeaderParts.serviceName})`
+    : '';
+}
+
+export function getTraceEmoji(spans: Span[]): string {
+  if (!spans.length) return '';
+
+  // prettier-ignore
+  const emojiSet = [
+    '🐶', '🐱', '🐭', '🦊', '🐨', '🐮', '🐷', '🐸', '🐵', '🐔', '🐤', '🦆',
+    '🦉', '🐝', '🦋', '🐢', '🦀', '🐳', '🐊', '🦒', '🪶', '🦩', '🐉', '🍄',
+    '🌸', '🌜', '⭐️', '🔥', '🌪️', '💧', '🍏', '🍊', '🍉', '🍒', '🥦', '🌽',
+    '🍠', '🥐', '🥖', '🥚', '🧀', '🍗', '🍟', '🍕', '🍣', '🍤', '🍙', '🍪',
+    '⚽️', '🏀', '🥎', '🎹', '🎲', '🎮', '🧩', '🚗', '🚲', '🛞', '🚂', '⛺️',
+    '💿', '📞', '⏰', '💡', '🔌', '💎', '🪚', '🧲', '💣', '🧬', '🔑', '🎀',
+    '📬', '🗑️', '📘', '📌', '🔍', '🔒', '🩷', '❗️', '🎵', '🔔', '🏴', '🚩', 
```

Comment:
some of these are quite similar to common UX elements, I would avoid them, e.g. '⭐️', '❗️', '🗑️', '📌', '🔍', '🔒', '💡'

Reply:
Sure.